tcms/testruns/tests/test_permissions.py

```diff
@@ -128,3 +128,31 @@ def verify_post_with_permission(self):
             self.assertEqual(case.history.latest().history_id, case_run.case_text_version)
             self.assertEqual(last_run.build, case_run.build)
             self.assertEqual(None, case_run.close_date)
+
+
+class GetTestRunViewTestCase(PermissionsTestCase):
+    permission_label = 'django_comments.add_comment'
+    http_method_names = ['get']
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.product = factories.ProductFactory(name='Kiwi')
+        cls.build = factories.BuildFactory(product=cls.product)
+        cls.test_run = factories.TestRunFactory(
+                                      build=cls.build)
+
+        cls.url = reverse('testruns-get', args=[cls.test_run.pk])
+        cls.post_data={"summary": "mock"}
+        super().setUpTestData()
+
+        cls.add_comment_html = \
+            '<a href="#" class="addBlue9 js-show-commentdialog">{0}</a>' \
+            .format(_('Add'))
+
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
+        self.assertContains(response, self.add_comment_html, html=True)
+
+    def verify_get_without_permission(self):
+        response = self.client.get(self.url)
+        self.assertNotContains(response, self.add_comment_html, html=True)
```

Comment:
in both verify methods you should assert that the testruns page is actually visible. Group those asserts into a helper method (e.g. assert_on_testrun_page()) and call them before you try to assert for presense of HTML snippet.